egs/rm/s5/local/chain/tuning/run_tdnn_wsj_rm_1a.sh

```diff
@@ -0,0 +1,192 @@
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+ali_dir=exp/tri3b_ali
+treedir=exp/chain/tri4_5n_tree
+lang=data/lang_chain_5n
+
+local/online/run_nnet2_common.sh  --stage $stage \
+                                  --ivector-dim 100 \
+                                  --nnet-affix "$nnet_affix" \
+                                  --mfcc-config $srcdir/conf/mfcc_hires.conf \
+                                  --extractor $srcdir/exp/nnet3/extractor || exit 1;
+
+if [ $stage -le 4 ]; then
+  # Get the alignments as lattices (gives the chain training more freedom).
+  # use the same num-jobs as the alignments
+  nj=$(cat exp/tri3b_ali/num_jobs) || exit 1;
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/train \
+    data/lang exp/tri3b exp/tri3b_lats
+  rm exp/tri3b_lats/fsts.*.gz # save space
+fi
+
+if [ $stage -le 5 ]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 6 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+    --leftmost-questions-truncate $leftmost_questions_truncate \
+    --cmd "$train_cmd" 1200 data/train $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 7 ]; then
+  echo "$0: creating neural net configs using the xconfig parser for";
+  echo "extra layers w.r.t source network.";
+  num_targets=$(tree-info $treedir/tree |grep num-pdfs|awk '{print $2}')
+  learning_rate_factor=$(echo "print 0.5/$xent_regularize" | python)
+  mkdir -p $dir
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  relu-renorm-layer name=tdnn7-target input=Append(tdnn6.renorm@-3,tdnn6.renorm@0) dim=$dim
+  ## adding the layers for chain branch
+  relu-renorm-layer name=prefinal-chain-target input=tdnn7-target dim=$dim target-rms=0.5
+  output-layer name=output-target include-log-softmax=false dim=$num_targets max-change=1.5
+  relu-renorm-layer name=prefinal-xent-target input=tdnn7-target dim=$dim target-rms=0.5
+  output-layer name=output-xent-target dim=$num_targets learning-rate-factor=$learning_rate_factor max-change=1.5
+EOF
+  # edits.config contains edits required to train transferred model.
```

Comment:
If we have xconfig-file containing new output-layers with same name 'output' and 'output-xent', it fails since we used more than one layer with same name!